### PR TITLE
Fix swallowed err in federation.

### PR DESCRIPTION
**Pull request by alrs**
I found a dropped error in the federation package.

### CODE REVIEW

**Comment by alrs** [comment]
I've signed the CLA.

File: federation/federation_configuration.go [hunk 1]
```diff
@@ -258,7 +258,9 @@ func (o *FederationConfiguration) EnsureConfiguration(c *fi.Context) error {
 
 		return s, nil
 	})
-
+	if err != nil {
+		return err
```
Comment (chrislovecnm) [inline federation/federation_configuration.go]: As before `fmt.Errorf` would we AWESOME

**Comment by alrs** [comment]
Thanks again, @chrislovecnm, I've been trying to mimic the existing err returns in the nearby code. I get it now, you guys want fmt.Errorf() every time. I've made the change. 

Hello to @kris-nova, thanks again for the Gophercon talk.

**Comment by justinsb** [comment]
Awesome, thank you @alrs .  Have you found a tool which can detect dropped errors?  If so maybe we should add it to our CI!

/lgtm

**Comment by justinsb** [comment]
(And yeah, we try to wrap errors unless we're very sure the error will be self explanatory.  I'm not sure it really matters here, but we almost always want to wrap errors that come back from 3rd party code because they tend to be cryptic - particular AWS errors)

**Pull request merged**